### DISCUSSION TITLE
How to implement non-linear protein binding function in PK-Sim/Mobi

### DISCUSSION THREAD

**Discussion by michielve** [Open-Systems-Pharmacology/Forum#1838] [Questions & Problems]
Hi all,

I am trying to implement a non-linear protein binding function in PK-Sim/Mobi. 
The following discussion was found on this topic, 
[Non linear Protein Binding Function in PK-Sim/Mobi #157
](https://github.com/Open-Systems-Pharmacology/Forum/issues/157)

In this discussion, the answer provided was:

> Basically, once you have set up your simulation in PKSim you export it to MoBi by "right-click" on the simulation and selecting -> "export to MoBi" in the drop-down menu.
> 
> The simulation will appear in MoBi and you can then edit the "fu" directly:
> 
> - Browse to your Building Blocks and double click on the Molecule Building Block (will have the sam name as your simulation)
> - On the right, a window will appear with a list of all your molecules in the simulation. Select (double click) the one for which you ant to change the fu.
> - Again, further to the right you will see two tabs with one named "Parameters". Select this one and browse to the parameter "Fraction unbound (plasma, reference value)".
> - There under "Value" change (with drop-down menu) from "Constant" to "Formula"
> - Enter the formula you want to use to describe your observed nonlinearity (please refer to the help menu ( the question mark on the very top right of the MoBi window) on how to create a formula
> - recreate your simulation --> run (also here the help menu may help)
> 

However, I can't figure out the correct formula to use. This is what I am trying to implement, with Concentration in molar.

`Concentration< 0.001 ? 0.10 : 0.10 + 5.4*Concentration`
where Concentration was defined as an Alias with Dimension Concentration (molar), however, **I can't seem to figure out which Path to use for the Concentration Alias.** 

The error is always that this alias references an entity with a path that cannot be found. I also could not find it in the manual/help menu.

Thanks in advance for any response

**Comment by StephanSchaller** [comment]
Dear Michiel,

thanks for your inquiry.

"fu" is a global parameter/property of a molecule, while "concentration" is "local", thus depending on the location of the molecule. You will not be able to use a local reference path definition (relative path) for the "concentration" in the formula you set up but would have to decide of a "representative" concentration value, here. So this could e.g. be either the arterial or venous blood plasma concentrations (use the full absolute path definition (global reference point)).

For your formula, please consider that fu can never be > 1, so I would use a saturatable function (Michaelis Menten or Hill equation) or a capped function, i.e., including "concentration > 1" condition:

'Concentration < 0.001 ? 0.10 : (Concentration > 1? 1 : 0.10 + 5.4*Concentration)'

Best,
Stephan

**Comment by michielve** [comment]
Thank you Stephan for your response. I got it working now by your help pointing me in the right direction.

The path to use that seems to work is:
Organism|VenousBlood|Plasma|MOLECULE|Concentration

The main issue was that the easy drag-and-drop menu of possible references objects was not visible and I could not scroll down further.
![image](https://github.com/user-attachments/assets/e68a9f68-8c00-44f9-a056-b37d61f1b05e)

Versus how it should look like after removing the notification view and widening the windows:
![image](https://github.com/user-attachments/assets/2fae769c-22d8-4bdb-87b1-61a558bd2254)

Without this menu it was very difficult to think of the right syntax to use in the path...

Thanks.

p.s. I fully agree with the issue on fu > 1, it was just a random formula that I typed

**Reply by StephanSchaller** [reply]
Happy this worked. Indeed, managing all the views and windows in MoBi becomes an issue if you do to use a quite large monitor. Recommend > 27 inch.
Happy this worked. If you follow up on a good refine equation for your nonlinear fu, would be great to post it here.

Thanks and good luck!
